### DISCUSSION TITLE
Support for plotting data (Bar charts/histograms)

### DISCUSSION THREAD

**Discussion by jaintj95** [Canner/WrenAI#620] [Ideas]
It would be great to have a feature where Wren automatically plots the data returned by the SQL query if the service determines the output data is plottable.

Step 1 -> User input 

Step 2 -> alongside generating sql code, agent returns a score between 0 and 1 determining if this input would lead to plottable data

Step 3 -> agent executes the sql query and returns data 

Step 4 -> this data is taken as input to further determine if it can be plotted using python and what's the best plot (bar chart/histogram/scatter plot) for this data

Step 5 -> add confidence scores from step 2 and step 4 to generate a final confidence score

Step 6 -> if the score is above a certain threshold, the agent will return python code to plot the data and display the plot/chart in UI

**Comment by wwwy3y3** [comment]
Hi @jaintj95 and everyone who's interested in this feature, we officially support chart generation starting from [0.13.2](https://github.com/Canner/WrenAI/releases/tag/0.13.2) 🎉

**Reply by wwwy3y3** [reply]
Learn more here: https://docs.getwren.ai/oss/guide/home/chart
